src/github.com/couchbase/sync_gateway/rest/handler.go

```diff
@@ -517,3 +499,23 @@ func (h *handler) writeStatus(status int, message string) {
 	jsonOut, _ := json.Marshal(db.Body{"error": errorStr, "reason": message})
 	h.response.Write(jsonOut)
 }
+
+// Returns the integer value of a URL query, restricted to a min and max value,
+// but returning 0 if missing or unparseable
+func getRestrictedIntQuery(values url.Values, query string, defaultValue, minValue, maxValue uint64) uint64 {
+	value := defaultValue
+	q := values.Get(query)
+	if q != "" {
+		var err error
+		value, err = strconv.ParseUint(q, 10, 64)
+		if err != nil {
+			value = 0
+		} else if value < minValue {
+			value = minValue
+		} else if value > maxValue && maxValue > 0 {
+			value = maxValue
+		}
+	}
+	log.Printf("Return val: %v", value)
+	return value
+}
```

Comment:
Ok I made this change and pushed a rebased commit.

The unit tests pass: http://factory.couchbase.com/view/build/view/mobile_dev/view/sync_gateway/view/master/job/build_sync_gateway_dev_cents-x64/130/console